src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/RazorSemanticTokensInfoService.cs

```diff
@@ -301,6 +400,37 @@ internal static bool TryGetMinimalCSharpRange(RazorCodeDocument codeDocument, Ra
         return response;
     }
 
+    private async Task<int[]?> GetMatchingCSharpResponseAsync(
```

Comment:
Again, this method looks like its pretty much a direct copy of the one above. Can we share?

If the only difference is whether the method gets an array of ranges, or a single range, then it seems like we could just unify it all together, and just pass an array with one element if the feature flag is off.